packages/opc-base/src/config/graphql.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  ApolloClient,
+  InMemoryCache,
+  createHttpLink,
+  split,
+  NormalizedCacheObject,
+} from "@apollo/client/core";
+import { setContext } from "@apollo/client/link/context";
+import { getMainDefinition } from "@apollo/client/utilities";
+import { WebSocketLink } from "@apollo/client/link/ws";
+
+import { GraphqlConfig } from "./types";
+
+/**
+ * Helper Class for making API Calls to the One Platform API Gateway
+ */
+export class APIService {
+  _apiBasePath: string;
+  _subscriptionsPath: string;
+  _wsLink: WebSocketLink;
+  apollo: ApolloClient<NormalizedCacheObject>;
+  constructor(config: GraphqlConfig) {
+    this._apiBasePath = config.apiBasePath;
+    this._subscriptionsPath = config.subscriptionsPath;
+    this._wsLink = new WebSocketLink({
+      uri: this._subscriptionsPath,
+      options: {
+        reconnect: true,
+        connectionParams: () => ({ ...this._headers }),
+      },
+    });
+    this.apollo = this.apolloInit();
+  }
+
+  private apolloInit() {
+    const httpLink = createHttpLink({
+      uri: this._apiBasePath,
+    });
+    const authLink = setContext((_, { headers }) => {
+      return {
+        headers: {
+          ...headers,
+          ...this._headers,
+        },
+      };
+    });
+
+    const link = split(
+      ({ query }) => {
+        const definition = getMainDefinition(query);
+        return (
+          definition.kind === "OperationDefinition" &&
+          definition.operation === "subscription"
+        );
+      },
+      this._wsLink,
+      authLink.concat(httpLink)
+    );
+    return new ApolloClient({
+      link,
+      cache: new InMemoryCache(),
```

Comment:
Add the config in opc-base to configure the cache

Reply:
Done